packages/artifact/src/internal/download-http-client.ts

```diff
@@ -263,26 +311,48 @@ export class DownloadHttpClient {
       if (isGzip) {
         const gunzip = zlib.createGunzip()
         response.message
+          .on('error', error => {
+            core.error(
+              `An error occurred while attempting to read the response stream`
+            )
```

Comment:
FYI, I was a little worried that calling `gunzip.close()` would trigger `on error` below and cause two core.errors to get logged. But I tried adding the following to the mock, and saw only this error block get hit:

```
          case 1:
            this.emit("error", new Error("injecting read error"))
            break
```